File: src/1748.sum-of-unique-elements.1848/solution.ts
function sumOfUnique (nums: number[]): number {
    const map = new Map<number, number>();
    let result = 0;
    for (let i = 0; i < nums.length; i++) {
        if (map.has(nums[i])) {
            if (map.get(nums[i])! === 1) {
                result -= nums[i];
                map.set(nums[i], 2);
            }
        } else {
            map.set(nums[i], 1);
            result += nums[i];
        }
    }
    return result;
}

File: src/1748.sum-of-unique-elements.1848/solution.java
class Solution {
    public int sumOfUnique(int[] nums) {
        int result = 0;
        Map<Integer, Long> map = Arrays.stream(nums).boxed()
                .collect(Collectors.groupingBy(Function.identity(), Collectors.counting()));
        for (Map.Entry<Integer, Long> entry : map.entrySet()) {
            if (entry.getValue() == 1) {
                result += entry.getKey();
            }
        }
        return result;
    }
}